### PR TITLE
Rework the release pipeline to use workspaces

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Stop using pipeline resources, use workspaces instead. This allows
us to use the catalog tasks for build and test too.
Add results with URLs of the published releases.

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>
/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
/meow
🎉

File: tekton/release-cheat-sheet.md [hunk 1]
```diff
@@ -12,64 +12,66 @@ the pipelines repo, a terminal window and a text editor.
...
-      --resource=builtControllerImage=controller-image \
-      --resource=builtWebhookImage=webhook-image \
-      --resource=builtDigestExporterImage=digest-exporter-image \
-      --resource=builtPullRequestInitImage=pull-request-init-image \
-      --resource=notification=post-release-trigger \
-    pipeline-release
+      --param=gitRevision="${TEKTON_RELEASE_GIT_SHA}" \
+      --param=serviceAccountPath=release.json \
+      --workspace name=release-secret,secret=release-secret \
+      --workspace name=workarea,volumeClaimTemplateFile=workspace-template.yaml
```
Comment (sbwsg) [inline tekton/release-cheat-sheet.md:51]: I didn't know tkn could do this, very cool!

**Review by sbwsg — APPROVED** [review]

File: tekton/release-pipeline.yaml [hunk 2]
```diff
@@ -8,139 +8,205 @@ spec:
...
+        - name: serviceAccountPath
+          value: $(params.serviceAccountPath)
+      workspaces:
+        - name: source
+          workspace: workarea
+          subpath: git
+        - name: release-secret
+          workspace: release-secret
     - name: publish-images
-      runAfter: [build, unit-tests]
```
Comment (pritidesai) [inline tekton/release-pipeline.yaml:106]: This is interesting, executing `publish-image` after `build` and `unit-test` are done and in parallel with `build-base-image`. `publish-image` is not dependent on `build-base-image` through `resources` as well (or am I missing it? 😜 ). Wondering how did this work until now?

**Comment by pritidesai** [comment]
/lgtm

Thanks @afrittoli 

Excited and a little nervous to release `0.22` 🤣

**Comment by afrittoli** [comment]
> /lgtm
> 
> Thanks @afrittoli 
> 
> Excited and a little nervous to release `0.22` 🤣 

Heh, it's now the same pipeline for nightly and full releases, so it will be tested quite a few times before it gets to v0.22 :)

**Pull request merged**